beacon-chain/sync/rpc_send_request.go

```diff
@@ -787,3 +815,14 @@ func downscorePeer(p2p p2p.P2P, peerID peer.ID, reason string, fields ...logrus.
 	newScore := p2p.Peers().Scorers().BadResponsesScorer().Increment(peerID)
 	log.WithFields(logrus.Fields{"peerID": peerID, "reason": reason, "newScore": newScore}).Debug("Downscore peer")
 }
+
+func DataColumnSidecarsByRangeRequest(columns []uint64, start, end primitives.Slot) *ethpb.DataColumnSidecarsByRangeRequest {
+	sort.Slice(columns, func(i, j int) bool {
+		return columns[i] < columns[j]
+	})
+	return &ethpb.DataColumnSidecarsByRangeRequest{
+		StartSlot: start,
+		Count:     uint64(end-start) + 1,
+		Columns:   columns,
+	}
```

Comment:
possible underflow here


```suggestion
	})
	count := uint64(end-start) + 1
	if end < start {
	   log.WithField("start", start).WithField("end", end).Error("Underflow in data column sidecar by range request construction. Defaulting to count = 1")
	   count = 1
	 }
	return &ethpb.DataColumnSidecarsByRangeRequest{
		StartSlot: start,
		Count:     count ,
		Columns:   columns,
	}
```

Reply:
This would be a bug in the caller so I would rather treat it as an error. I'll add that check.